### PR TITLE
Include completions for items from the same file; Clean up completions generation for callables

### LINKED ISSUE #1862: Callables from same file don't show up in the completions list

### PR DESCRIPTION
**Pull request by sezna**
This PR closes #1862.

I had previously touched this function in #1820. I did my best to port it and maintain existing functionality, but there was one conditional that I messed up, and it was around callables originating from the same file.

This PR fixes the bug that caused. But also, this PR tries to simplify `callable_decl_to_completion_item` so this is less likely to happen in the future.

### CODE REVIEW

File: language_service/src/completion.rs
Comment (swernli) [inline language_service/src/completion.rs:785]: nitpick: why the spaces between these comments? It seems like the could be all one paragraph with some punctuation between then. Not critical to functionality, of course.

**Review by swernli — APPROVED** [review]

File: language_service/src/completion.rs
Comment (sezna) [reply language_service/src/completion.rs]: I was just using newlines to separate my sentences. I've updated it to use the more canonical periods-based sentence structure that English provides 😄

File: language_service/src/completion.rs
Comment (billti) [inline language_service/src/completion.rs:770]: I find this function borderline unreadable, but as it was before-hand, will approve to unblock. We should really focus on code clarity and readability more when authoring though, as it makes it really hard to feel confident that you would have spotted any logic bugs when reviewing.

**Review by billti — APPROVED** [review]

File: language_service/src/completion.rs
Comment (sezna) [reply language_service/src/completion.rs:770]: I have been working to incrementally improve this, but you're right, it could use a deeper refactor. I filed #1864 to track this.

**Pull request merged**
